### DISCUSSION TITLE
When will Panda CSS reach a stable release?

### DISCUSSION THREAD

**Discussion by land-cap** [chakra-ui/panda#2487] [Q&A] [answered]
Panda CSS is probably the best thing that happened to CSS-in-JS since styled-components and it's great. I use it for side projects and love it but I'm curious to know when will it reach a stable 1.0 release. Right now it's hard to convince everyone to commit to using it in production given it's not stable yet... Am I right to assume after integrating it with Chakra it will become stable?

Thank you again for this great tool @segunadebayo and the whole team!

**Comment by segunadebayo** [comment]
Thanks for your remark @land-cap.

We're pretty close to the v1.0. This is independent of the release of Chakra UI v3.

There are a few PRs (related to the studio and presets) left to land before a v1 is released. Most of the features we have in mind for Panda are already in.

**Reply by land-cap** [reply]
Thank you for the quick reply. Looking forward to see studio! Have been searching a lot and havent found how to set it up yet. Stoked to finally see it🔥

**Reply by land-cap** [reply]
Hi @segunadebayo!
I wanted to ask how close is the 1.0 release? 🙂 
Will the API change much or would you say it's better not to use it in production code?
Thank you!

**Reply by kris-ellery** [reply]
Hello @segunadebayo! Would you mind sharing an update on the Panda CSS 1.0 roadmap? We're actively using it at Zillow for our design system, and we've already deployed it on some production pages. Thanks!

**Reply by segunadebayo** [reply, marked as answer]
v1 is going live next week or so.

Panda has been relatively stable for a while, and we're happy with it. We'll think through any tiny updates over the weekend.

**Reply by cgatian** [reply]
@segunadebayo I looked into #3089 and think it's a pretty big change to how Panda generates the CSS. The team might want to review this issue and #3137 when considering 1.0 release.

**Comment by mrhoodz** [comment]
Great news 🐼

**Comment by kris-ellery** [comment]
@segunadebayo Hoping to gently reopen the conversation about version 1.0. I know this was marked as resolved, but since it hasn't shipped yet, it's still a blocking issue for many of us. Would it be possible to get a revised timeline and a quick update on what's standing in the way of the release? Thanks in advance for your help.

**Reply by cgatian** [reply]
We actually couldn't wait any longer and have moved off of Panda. 😢

**Reply by jimmymorris** [reply]
What did you transition to?

**Reply by cgatian** [reply]
We shoehorned Panda in for about 8 months but the lack of contributors, and a 1.0 release ultimately ended our evaluation.

Other issues we had trouble swallowing:
* No dynamic variables 
* Keyframes must be defined in global config (painful)
* Unintuitive configuration for teams that have never used Panda (each project needed to have an intimate understanding of panda config, no one wants to maintain a list of components they use in a separate configuration, it should just be a part of the bundler resolution)
* Style extraction issue #3089. (Your stylesheet will contain classes and values that are completely unrelated to Panda)
* Complex monorepo support (multiple projects utilizing the component library)
* No tree shaking styles
* First class bundler support. (When things get rough, the suggested path is to use the Panda CLI, which essentially means spawn a separate thread and manage style generation in a separate process. This is painful and can cause double reloads during development.)
* Maintainers constantly moving issues to discussions, which was extremely frustrating at first but now seems as a dubious practice to hide the library's flaws

Panda's great for a standalone project, where you understand Panda and own the pipeline. Soon as you drop it into a monorepo where individual contributors need to be aware of how Panda works, it falls apart.

Our code base is enormous and it uses Emotion. We took the best parts we liked about Panda and implemented it in Emotion. The CSS style generation is now handled by style dictionary, and it actually all came together relatively quickly, and ultimately lead to better DX, at the cost of losing atomic styles. And I'm not convinced automatic atomic styles is the right path. We had some components with over 25 classes added to them, it really started getting painful. 

With all that said, Panda is a great library. The authors really did a tremendous job. We were sad when we decided to pull the plug.

**Reply by segunadebayo** [reply]
Thanks for the feedback @cgatian. We'll definitely improve Panda's monorepo DX over time.

**Comment by segunadebayo** [comment]
We've now released Panda v1
https://github.com/chakra-ui/panda/discussions/3321
